include/seqan/basic/allocator_interface.h

```diff
@@ -204,6 +204,7 @@ allocate(T const &,
     if (data)
         SEQAN_PROADD(SEQAN_PROMEMORY, count * sizeof(TValue));
 #endif
+	memset((void*)data, 0, count * sizeof(TValue));
```

Comment:
Hmm the problem must be somewhere above... allocate() does not have to initialize the memory.

See http://docs.seqan.de/seqan/develop/class_Allocator.html#Allocator#allocate:
"These objects are not constructed by allocate."